File: app/src/main/java/vn/edu/utt/uttqlsv/controller/utils/Validator.java
package vn.edu.utt.uttqlsv.controller.utils;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class Validator {
    private static final String PASSWORD_PATTERN = "^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#&()–[{}]:;',?/*~$^+=<>]).{8,20}$";
    private static final String FLOATING_NUMBER_PATTERN = "[+-]?[0-9]+(\\\\.[0-9]+)?([Ee][+-]?[0-9]+)?";
    private static final String DATE_PATTERN = "^\\d{2}/\\d{2}/\\d{4}";

    public static boolean isValidPassword(String password) {
        Pattern pattern = Pattern.compile(PASSWORD_PATTERN);
        Matcher matcher = pattern.matcher(password);
        return matcher.matches();
    }

    public static boolean isValidFloatingNumber(String f) {
        Pattern pattern = Pattern.compile(FLOATING_NUMBER_PATTERN);
        Matcher matcher = pattern.matcher(f);
        return matcher.matches();
    }

    public static boolean isValidDateString(String date) {
        Pattern pattern = Pattern.compile(DATE_PATTERN);
        Matcher matcher = pattern.matcher(date);
        return matcher.matches();
    }

}

File: app/src/main/java/vn/edu/utt/uttqlsv/controller/utils/Validator.kt
package vn.edu.utt.uttqlsv.controller.utils

import java.util.regex.Pattern

object Validator {
    //ReGex Pattern for password, floating number, date checking
    private const val PASSWORD_PATTERN =
        "^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#&()–[{}]:;',?/*~$^+=<>]).{8,20}$"

    private const val FLOATING_NUMBER_PATTERN = "[+-]?[0-9]+(\\.[0-9]+)?([Ee][+-]?[0-9]+)?"

    private const val DATE_PATTERN = "^\\d{2}/\\d{2}/\\d{4}"

    fun isValidPassword(password: String): Boolean {
        val pattern = Pattern.compile(PASSWORD_PATTERN)
        val matcher = pattern.matcher(password)
        return matcher.matches()
    }

    /**
     * This function will validate floating number passed as string is valid or not
     * @param float floating number as string
     * @return true or false
     */
    fun isValidFloatingNumber(float: String): Boolean {
        val pattern = Pattern.compile(FLOATING_NUMBER_PATTERN)
        val matcher = pattern.matcher(float)
        return matcher.matches()
    }

    /**
     * this function will check date string is inform dd/MM/yyyy
     * @param date date string
     * @return true or false
     */
    fun isValidDateString(date: String): Boolean {
        val pattern = Pattern.compile(DATE_PATTERN)
        val matcher = pattern.matcher(date)
        return matcher.matches()
    }
}